src/vs/editor/contrib/codeAction/browser/codeActionCommands.ts

```diff
@@ -320,12 +324,48 @@ export class RefactorAction extends EditorAction {
 					: nls.localize('editor.action.refactor.noneMessage', "No refactorings available"),
 			{
 				include: CodeActionKind.Refactor.contains(args.kind) ? args.kind : CodeActionKind.None,
-				onlyIncludePreferredActions: args.preferred,
+				onlyIncludePreferredActions: args.preferred
 			},
 			args.apply);
 	}
 }
 
+export class RefactorPreview extends EditorAction {
+
+	constructor() {
+		super({
+			id: refactorPreviewCommandId,
+			label: nls.localize('refactor.preview.label', "Refactor Preview..."),
```

Comment:
Let's give this a better name to make it more clear what the action does. Maybe `Refactor with preview`?